spyder/config/user.py

```diff
@@ -397,6 +398,17 @@ def get(self, section, option, default=NoDefault):
                 value = ast.literal_eval(value)
             except (SyntaxError, ValueError):
                 pass
+        
+        # It's necessary to verify if the window/position value is valid 
+        # with the current screen. See issue 3748
+        if option == 'window/position':
+            width = value[0]
+            height = value[1]
+            screenShape = QApplication.desktop().geometry()
+            current_width = screenShape.width()
+            current_height = screenShape.height()
+            if current_width < width or current_height < height:
+                value = self.get_default(section, option)

```

Comment:
Although this is the right thing to do, this is not the right place. You need where this setting is applied (most probably in `app/mainwindow.py`) and use this validation there.